core/src/main/java/org/sonarsource/sonarlint/core/tracking/IndexedObjectStore.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * SonarLint Core - Implementation
+ * Copyright (C) 2009-2016 SonarSource SA
+ * mailto:contact AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.sonarlint.core.tracking;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collection;
+import java.util.Optional;
+import org.sonarsource.sonarlint.core.client.api.connected.objectstore.ObjectStore;
+import org.sonarsource.sonarlint.core.client.api.connected.objectstore.PathMapper;
+import org.sonarsource.sonarlint.core.client.api.connected.objectstore.Reader;
+import org.sonarsource.sonarlint.core.client.api.connected.objectstore.Writer;
+
+/**
+ * An ObjectStore without internal cache that derives the filesystem path to storage using a provided PathMapper.
+ *
+ * @param <K> type of the key to store by and used when reading back; must be hashable
+ * @param <V> type of the value to store
+ */
+class IndexedObjectStore<K, V> implements ObjectStore<K, V> {
+  private final Logger logger;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Make the "logger" logger private static final and rename it to comply with the format "LOG(?:GER)?". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1312)